src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -258,7 +258,7 @@ SGVector<T> from_gpu(const SGVector<T>& vec)
 			SG_SERROR("Data memory on GPU but no GPU backend registered. \
 						This can happen if the GPU backend was de-activated \
 						after memory has been transferred to GPU.\n");
-			return NULL;
+			return false;
```

Comment:
okay I didn't find exact same scenario in Shogun  (because it seems not achievable) but I can make some changes:
- `SGVector<T>* from_gpu() { return NULL; }`
or
-   `SGVector<T> from_gpu() { return SGVector<T>(); }`
or
-  Use `boost::optional`